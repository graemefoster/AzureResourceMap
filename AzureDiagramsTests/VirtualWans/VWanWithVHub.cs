using AzureDiagrams.Resources;
using Shouldly;

namespace AzureDiagramsTests.VirtualWans;

public class VWanWithVHub
{
    [Fact]
    public async Task CanDrawDiagram()
    {
        var resources = BuildScenario();

        var diagram = await AzureDiagramGenerator.DrawIoDiagramGenerator.DrawDiagram(
            resources,
            false,
            false,
            false,
            true,
            false);

        diagram.ShouldMatchApproved();
    }

    private static AzureResource[] BuildScenario()
    {
        var vwanId = new Guid("6d4ced93-d694-4e0c-b3a0-6cef2c095061").ToString();
        var vnetId = "/subscriptions/123/resourceGroups/rg1/providers/microsoft.network/virtualnetworks/vnet123";

        var vwan = new VWan(vwanId, "vwan");

        var vnet = new VNet(vnetId, "test-vnet", Array.Empty<VNet.Subnet>());

        var vhub = new VHub(
            vwanId, 
            new Guid("318e3768-dacb-4e2c-b87e-288107259a6c").ToString(), 
            "Hub-1",
            new[] { vnetId }
            );

        // var appServicePlanId =
        //     "/subscriptions/123/resourceGroups/rg1/providers/Microsoft.Web/serverFarms/TestAppServicePlan";
        //
        // var asp = new AppServicePlan(appServicePlanId, "TestAppServicePlan");
        //
        // var peId = new Guid("36a566f9-f6b8-439c-9371-204842160e2b").ToString();
        // var pe2Id = new Guid("36a566f9-f6b8-439c-9371-204842160e2d").ToString();
        //
        // var app1HostName = "TestApp1.azurewebsites.net";
        // var app2HostName = "TestApp2.azurewebsites.net";
        //
        // var app1 = new AppServiceApp("/subscriptions/123/resourceGroups/rg1/providers/Microsoft.Web/sites/TestApp1",
        //     appServicePlanId,
        //     "TestApp1",
        //     false,
        //     Array.Empty<string>(),
        //     new[] { app1HostName },
        //     virtualNetworkSubnetId: $"{vnetId}/subnets/vnet-integration",
        //     new PrivateEndpointExtensions(new[] { peId }));
        //
        // var app2 = new AppServiceApp("/subscriptions/123/resourceGroups/rg1/providers/Microsoft.Web/sites/TestApp2",
        //     appServicePlanId,
        //     "TestApp2",
        //     false,
        //     new[] { $"https://{app1HostName}" },
        //     new[] { app2HostName },
        //     virtualNetworkSubnetId: $"{vnetId}/subnets/vnet-integration",
        //     new PrivateEndpointExtensions(new[] { pe2Id }));
        //
        // var nicId = new Guid("36a566f9-f6b8-439c-9371-204842160e2c").ToString();
        // var nic = new Nic(nicId,
        //     IpConfigurations.ForPrivateEndpoint("10.1.1.1", $"{vnetId}/subnets/private-endpoints", app1HostName));
        //
        // var privateEndpoint = new PrivateEndpoint(peId, new[] { $"{vnetId}/subnets/private-endpoints" },
        //     new[] { nic.Id }, new[] { $"https://{app1HostName}" });
        //
        // var nic2Id = new Guid("36a566f9-f6b8-439c-9371-204842160e2e").ToString();
        // var nic2 = new Nic(nic2Id,
        //     IpConfigurations.ForPrivateEndpoint("10.1.1.2", $"{vnetId}/subnets/private-endpoints", app2HostName));
        //
        // var privateEndpoint2 = new PrivateEndpoint(pe2Id, new[] { $"{vnetId}/subnets/private-endpoints" },
        //     new[] { nic2.Id }, new[] { $"https://{app2HostName}" });

        var azureResources = new AzureResource[]
        {
            vwan,
            vhub,
            vnet
        };

        var resources = azureResources.Process();
        return resources;
    }
}