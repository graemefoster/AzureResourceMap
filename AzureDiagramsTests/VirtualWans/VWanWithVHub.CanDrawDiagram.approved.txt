<mxGraphModel>
	<root>
		<mxCell id="0" />
		<mxCell id="1" parent="0" />
<mxCell id="823cbfb3-2d33-d09b-d582-df5b793b61f4" value="vwan" style="html=1;image;image=img/lib/azure2/networking/Virtual_WANs.svg;fontSize=12;labelPosition=bottom" vertex="1" parent="823cbfb3-2d33-d09b-d582-df5b793b61f4.container">
    <mxGeometry x="300" y="50" width="50" height="50" 
    as="geometry" />
</mxCell>
<mxCell id="4b781b88-7e3d-9462-d7a0-69670644f6fe" value="Hub-1&lt;br/&gt;(VHub)" style="rounded=0;whiteSpace=wrap;html=1;fillColor=#d5e8d4;verticalAlign=middle" vertex="1" parent="823cbfb3-2d33-d09b-d582-df5b793b61f4.container">
    <mxGeometry x="50" y="50" width="150" height="75" 
    as="geometry" />
</mxCell>
<mxCell id="bd3a1a44-a441-608a-7d61-14ee6b5f190c.subnets.empty" value="Empty&lt;br/&gt;(Subnet)" style="rounded=0;whiteSpace=wrap;html=1;fillColor=#ffffff;verticalAlign=middle" vertex="1" parent="bd3a1a44-a441-608a-7d61-14ee6b5f190c">
    <mxGeometry x="50" y="50" width="150" height="75" 
    as="geometry" />
</mxCell>
<mxCell id="6278f6e0-78c3-8d53-0626-4965e13bdaa4" value="Virtual Hub Link" style="html=1;image;image=img/lib/azure2/networking/Virtual_WANs.svg;fontSize=12;labelPosition=bottom" vertex="1" parent="bd3a1a44-a441-608a-7d61-14ee6b5f190c">
    <mxGeometry x="300" y="50" width="50" height="50" 
    as="geometry" />
</mxCell>
<mxCell id="823cbfb3-2d33-d09b-d582-df5b793b61f4.container" value="vwan&lt;br/&gt;(VWan)" style="rounded=0;whiteSpace=wrap;html=1;fillColor=#dae8fc;verticalAlign=top" vertex="1" parent="207c67fc-34bc-d9f0-369e-0f16975fc114.container">
    <mxGeometry x="50" y="50" width="400" height="185" 
    as="geometry" />
</mxCell>
<mxCell id="bd3a1a44-a441-608a-7d61-14ee6b5f190c" value="test-vnet&lt;br/&gt;(VNet)" style="rounded=0;whiteSpace=wrap;html=1;fillColor=#F8CECC;verticalAlign=top" vertex="1" parent="207c67fc-34bc-d9f0-369e-0f16975fc114.container">
    <mxGeometry x="50" y="260" width="400" height="185" 
    as="geometry" />
</mxCell>
<mxCell id="bd3a1a44-a441-608a-7d61-14ee6b5f190c.image.0" style="html=1;image;image=img/lib/azure2/networking/Virtual_Networks.svg;fontSize=12;labelPosition=bottom" vertex="1" parent="bd3a1a44-a441-608a-7d61-14ee6b5f190c">
    <mxGeometry x="360" y="145" width="30" height="30" 
    as="geometry" />
</mxCell>
<mxCell id="207c67fc-34bc-d9f0-369e-0f16975fc114.container" value="global&lt;br/&gt;(Region)" style="rounded=0;whiteSpace=wrap;html=1;fillColor=#F5F5F5;verticalAlign=top" vertex="1" parent="1">
    <mxGeometry x="0" y="0" width="500" height="505" 
    as="geometry" />
</mxCell>
<mxCell id="c5131159-6fd2-a9d3-d3a0-10808df28731" 
         style="edgeStyle=orthogonalEdgeStyle;rounded=0;orthogonalLoop=1;jettySize=auto;html=1;" edge="1" parent="1" 
         source="6278f6e0-78c3-8d53-0626-4965e13bdaa4" target="4b781b88-7e3d-9462-d7a0-69670644f6fe">
            <mxGeometry relative="1" as="geometry" />
             </mxCell>
             

	</root>
</mxGraphModel>